### PR TITLE
add api-logs-limit flag

### PR DESCRIPTION
**Pull request by libotony**
# Description

Add a flag to allow node runner to specify the logs limit for the logs filter API.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also
list any relevant details for your test configuration

- [x] Test A
- [x] Test B

**Test Configuration**:

* Go Version:
* Hardware:
* Docker Version:

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] New and existing E2E tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have not added any vulnerable dependencies to my code

### CODE REVIEW

**Comment by darrenvechain** [comment]
Should we return an API error if the user specify more than the limit? Because if the user requests 5000 and gets 1000, then it looks like there are no more results, and they will stop making paginated requests

File: cmd/thor/flags.go [hunk 1]
```diff
@@ -59,6 +59,11 @@ var (
 		Name:  "api-allow-custom-tracer",
 		Usage: "allow custom JS tracer to be used tracer API",
 	}
+	apiLogsLimitFlag = cli.IntFlag{
```
Comment (leszek-vechain) [inline cmd/thor/flags.go:62]: should we add it too docs, or it is hidden ?

Comment (libotony) [reply cmd/thor/flags.go:62]: Will add to the docs, thanks!

**Review by otherview — COMMENTED** [review]
Structure is solid, and the limit makes sense to happen at the api lvl.

Probably worth to have a unit test that ensures the filtering is returning the max number of results in the `events_test.go` and in the `transfers_test.go`. We're testing the `NormalizeOptions` but if for example that function gets left out for some reason the tests won't capture that the query becomes now unbounded ?

Also, perhaps update the api yml definition to account for the existence of the flag ?

**Comment by darrenvechain** [comment]
@libotony I made a PR here to fix the E2E. We need to merge this to main and then put the commit SHA in the e2e GHA

https://github.com/vechain/thor-e2e-tests/pull/24/files

**Review by otherview — APPROVED** [review]

**Pull request merged**